### PR TITLE
Updated the CLI to allow for a filename option to override the default 'routeType.ts' filename 

### LINKED ISSUE #112: Possible to allow the setting of a filename in the cli to be used in place of routeType.ts?
I have no idea how hard or easy it would be to do this, but I think it would be great if we had the option to set the filename this tool looks for from the CLI or something. It would be nice to be able to name my file route-type.ts instead of routeType.ts to match the rest of my app.

### PR DESCRIPTION
**Pull request by JasonPaff**
closes #112 

## This PR:

- updates the cli to allow for a custom filename to be set under the --filename flag to replace the default of 'routeType.ts'
- updates the examples folder with an example project that uses a custom filename (route-type.ts) for the routeType.ts files
- updates the options section of the docs site to include --filename

 - [ ] (if ready to be merged) Yes I have made a changeset

### CODE REVIEW

File: examples/appdir-custom-filename/_next-typesafe-url_.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// prettier-ignore
+/* eslint-disable */
+
+
+declare module "@@@next-typesafe-url" {
+  import type { InferRoute, StaticRoute } from "next-typesafe-url";
+  
+  interface DynamicRouter {
+    "/foo/[id]/nest": InferRoute<import("./src/app/(test)/foo/[id]/nest/routeType").RouteType>;
+    "/foo/[id]": InferRoute<import("./src/app/(test)/foo/[id]/routeType").RouteType>;
```
Comment (ethanniser) [inline examples/appdir-custom-filename/_next-typesafe-url_.d.ts:14]: these dont look right?

**Review by ethanniser — COMMENTED** [review]
sorry im super late to this- need to fix my github notifications

thank you for the PR, this looks great except for the example does not seem to actually have the right generated file

if that is fixed I will merge

thanks again

**Comment by JasonPaff** [comment]
> sorry im super late to this- need to fix my github notifications
> 
> thank you for the PR, this looks great except for the example does not seem to actually have the right generated file
> 
> if that is fixed I will merge
> 
> thanks again

Sorry about that, looks like I didn't actually commit the last set of changes before opening the PR. That is now fixed!

**Comment by ethanniser** [comment]
Just stripped down the example to remove duplication and fixed some formatting

Thanks for the pr

**Pull request merged**